mbed_host_tests/host_tests_conn_proxy/conn_proxy.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python
+"""
+mbed SDK
+Copyright (c) 2011-2015 ARM Limited
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import re
+import sys
+import uuid
+from serial import Serial, SerialException
+from time import sleep, time
+from conn_proxy_logger import HtrunLogger
+
+
+class SerialConnectorPrimitive(object):
+    def __init__(self, port, baudrate, prn_lock, config):
+        #ConnectorPrimitive.__init__(self)
+        self.prn_lock = prn_lock
+        self.config = config
+        self.logger = HtrunLogger(prn_lock)
+        program_cycle_s = int(self.config['program_cycle_s']) if 'program_cycle_s' in self.config else 4
+
+        self.logger.prn_inf("Serial port=%s baudrate=%s program_cycle_s=%s"% (str(port), (baudrate), (program_cycle_s)))
+        try:
+            self.serial = Serial(port, baudrate=baudrate, timeout=0.2)
+        except SerialException as e:
+            self.logger.prn_err(str(e))
+            self.serial = None
```

Comment:
if Serial() fails , does it make sense go forward and do send_break and program_cycle_wait() ? send_break does not make sense if serial is None. program cycle might stay.